source/data-formats/dates-and-times.txt

```diff
@@ -0,0 +1,448 @@
+   current local time.
+   
+   Instead, always call the ``datetime.now(tz=datetime.timezone.utc)`` method, which
+   returns the current time in UTC.
+
+.. _pymongo-out-of-range-datetimes:
+
+Handling Out-of-Range datetimes
+-------------------------------
+
+Python's ``datetime`` class can represent only ``datetime`` values between
+``datetime.min`` and ``datetime.max`` (years 1-9999).
+You can represent a much greater range of dates and times by using BSON, which allows any
+64-bit millisecond value from the Unix epoch. 
+
+To represent a BSON time with {+driver-short+}, create a
+``datetime_ms.DatetimeMS`` object, a wrapper for Python's built-in ``int`` type.
+You can manually create a ``DatetimeMS`` object by passing in one of the following values:
+
+- An ``int`` representing the number of milliseconds since the Unix epoch
+- A ``datetime`` object
+
+The following code example constructs a ``DatetimeMS`` object by passing in an ``int``
+value that represents the year -146136543, a date outside of the ``datetime`` range: 
+
+.. code-block:: python
+
+     from bson.datetime_ms import DatetimeMS
+
+     out_of_range = DatetimeMS(-(2**62))
+
+You can also instruct {+driver-short+} to automatically decode UTC ``datetime`` values as
+``DatetimeMS`` objects. To do so, set the ``datetime_conversion`` parameter of ``CodecOptions``
+to a value from the ``datetime_ms.DatetimeConversion`` enum. The following sections
+describe these values.
+
+.. tip::
+
+   For more information about the Unix epoch, see its :wikipedia:`Wikipedia <Unix_time>`
+   entry.
+
+DatetimeConversion.DATETIME
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``DatetimeConversion.DATETIME`` is the default value. This value causes {+driver-short+}
+to raise an error when it tries to decode an out-of-range date, as shown
+in the following example:
+
+.. io-code-block::
+
+   .. input::
+      :language: python
+
+      from datetime import datetime
+      from bson import encode, decode
+      from bson.datetime_ms import DatetimeMS
+      
+      out_of_range = DatetimeMS(-(2**62))
+      val = encode({"date": out_of_range})
+      decoded = decode(val)
+      print(decoded)
+
+   .. output::
+      :language: shell
+
+      ...
+      bson.errors.InvalidBSON: year -146136543 is out of range (Consider Using
+      CodecOptions(datetime_conversion=DATETIME_AUTO) or
+      MongoClient(datetime_conversion='DATETIME_AUTO')).
+      ...
+
+DatetimeConversion.DATETIME_MS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This value instructs {+driver-short+} to return only ``DatetimeMS`` objects, even if
+the date is within the range of ``datetime``:
+
+.. io-code-block::
+
+   .. input::
+      :language: python
+
+      from datetime import datetime
+      from bson import encode, decode
+      from bson.datetime_ms import DatetimeMS
+      from bson.codec_options import CodecOptions, DatetimeConversion
+   
+      val = encode({"date": datetime(1970, 1, 1)})
```

Comment:
Nitpick, but seeing the output converting this to `DatetimeMS(0)` for some reason made me assume it was an out of range datetime and therefore showing as 0 (which after reading the later section I realized I thought this was "clamping". 
I think that could be avoided by adjusting the date a bit so it doesn't fall right on the epoch